crates/ra_lsp_server/src/world.rs

```diff
@@ -279,3 +281,65 @@ impl WorldSnapshot {
         self.analysis.feature_flags()
     }
 }
+
+/// Returns a `Url` object from a given path, will lowercase drive letters if present.
+/// This will only happen when processing windows paths.
+///
+/// When processing non-windows path, this is essentially the same as `Url::from_file_path`.
+fn url_from_path_with_drive_lowercasing(path: impl AsRef<Path>) -> Result<Url> {
+    let component_has_windows_drive = path
+        .as_ref()
+        .components()
+        .find(|comp| {
+            if let Component::Prefix(c) = comp {
+                match c.kind() {
+                    Prefix::Disk(_) | Prefix::VerbatimDisk(_) => return true,
+                    _ => return false,
+                }
+            }
+            false
+        })
+        .is_some();
+
+    // VSCode expects drive letters to be lowercased, where rust will uppercase the drive letters.
+    if component_has_windows_drive {
+        let url_original = Url::from_file_path(&path)
+            .map_err(|_| format!("can't convert path to url: {}", path.as_ref().display()))?;
+
+        let drive_partition: Vec<&str> = url_original.as_str().rsplitn(2, ':').collect();
+
+        // There is a drive partition, but we never found a colon.
+        // This should not happen, but in this case we just pass it through.
+        if drive_partition.len() == 1 {
+            return Ok(url_original);
+        }
+
+        let joined = drive_partition[1].to_ascii_lowercase() + ":" + drive_partition[0];
+        let url = Url::from_str(&joined).expect("This came from a valid `Url`");
+
+        Ok(url)
+    } else {
+        Ok(Url::from_file_path(&path)
+            .map_err(|_| format!("can't convert path to url: {}", path.as_ref().display()))?)
+    }
+}
+
+// `Url` is not able to parse windows paths on unix machines.
```

Comment:
Ouch, this is not at all pretty. I don't really understand why an `URL` is a platform-specific thing, but maybe there's a good reason for this...